protocol/contracts/beanstalk/silo/ConvertFacet.sol

```diff
@@ -43,119 +65,307 @@ contract ConvertFacet is ReentrancyGuard {
+     * given that the conversion is supported by the ConvertFacet.
+     * For example, a user can convert LP into Bean, only when beanstalk is below peg, 
+     * or convert beans into LP, only when beanstalk is above peg.
+     * @param convertData  input parameters to determine the conversion type.
+     * @param stems the stems of the deposits to convert 
+     * @param amounts the amounts within each deposit to convert
+     * @return toStem the new stems of the converted deposit
+     * @return fromAmount the amount of tokens converted from
+     * @return toAmount the amount of tokens converted to
+     * @return fromBdv the bdv of the deposits converted from
+     * @return toBdv the bdv of the deposit converted to
+     */
     function convert(
         bytes calldata convertData,
-        uint32[] memory crates,
+        int96[] memory stems,
         uint256[] memory amounts
     )
         external
         payable
         nonReentrant
-        returns (uint32 toSeason, uint256 fromAmount, uint256 toAmount, uint256 fromBdv, uint256 toBdv)
+        returns (int96 toStem, uint256 fromAmount, uint256 toAmount, uint256 fromBdv, uint256 toBdv)
     {
-        LibInternal.updateSilo(msg.sender);
-
         address toToken; address fromToken; uint256 grownStalk;
         (toToken, fromToken, toAmount, fromAmount) = LibConvert.convert(
             convertData
         );
 
+        LibSilo._mow(msg.sender, fromToken);
+        LibSilo._mow(msg.sender, toToken);
+
         (grownStalk, fromBdv) = _withdrawTokens(
             fromToken,
-            crates,
+            stems,
             amounts,
             fromAmount
         );
 
         uint256 newBdv = LibTokenSilo.beanDenominatedValue(toToken, toAmount);
         toBdv = newBdv > fromBdv ? newBdv : fromBdv;
 
-        toSeason = _depositTokens(toToken, toAmount, toBdv, grownStalk);
+        toStem = _depositTokensForConvert(toToken, toAmount, toBdv, grownStalk);
 
         emit Convert(msg.sender, fromToken, toToken, fromAmount, toAmount);
     }
 
+    //////////////////////// UPDATE UNRIPE DEPOSITS ////////////////////////
+
+    /**
+     * @notice Update the BDV of an Unripe Deposit. Allows the user to claim
+     * Stalk as the BDV of Unripe tokens increases during the Barn
+     * Raise. This was introduced as a part of the Replant.
+     *
+     * @dev Should revert if `ogBDV > newBDV`. A user cannot lose BDV during an
+     * Enroot operation.
+     *
+     * Gas optimization: We neglect to check if `token` is whitelisted. If a
+     * token is not whitelisted, it cannot be Deposited, and thus cannot be Removed.
+     * 
+     * {LibTokenSilo-removeDepositFromAccount} should revert if there isn't
+     * enough balance of `token` to remove.
+     * Because the amount and the stem of an Deposit does not change, 
+     * an ERC1155 event does not need to be emitted.
+     */
+    function enrootDeposit(
```

Comment:
I feel like the convert facet is fine since it's basically just a convert, but if you feel it's better to move it to another one, happy to do so. @Brean0 what do you think?